From e39f35bb07995da8d5c735011501724a8f1af05e Mon Sep 17 00:00:00 2001
From: mrkirby153 <mr.austinwhyte@gmail.com>
Date: Tue, 23 Aug 2022 23:14:39 -0700
Subject: [PATCH] dmenu border patch

---
 config.def.h | 3 +++
 dmenu.c      | 3 ++-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/config.def.h b/config.def.h
index 4fe84dc..7308bae 100644
--- a/config.def.h
+++ b/config.def.h
@@ -24,3 +24,6 @@ static unsigned int lines      = 0;
  * for example: " /?\"&[]"
  */
 static const char worddelimiters[] = " ";
+
+/* Size of the window border */
+static const unsigned int border_width = 5;
diff --git a/dmenu.c b/dmenu.c
index 7476e43..5fab736 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -769,9 +769,10 @@ setup(void)
 	swa.override_redirect = True;
 	swa.background_pixel = scheme[SchemeNorm][ColBg].pixel;
 	swa.event_mask = ExposureMask | KeyPressMask | VisibilityChangeMask;
-	win = XCreateWindow(dpy, parentwin, x, y, mw, mh, 0,
+	win = XCreateWindow(dpy, parentwin, x, y, mw, mh, border_width,
 	                    CopyFromParent, CopyFromParent, CopyFromParent,
 	                    CWOverrideRedirect | CWBackPixel | CWEventMask, &swa);
+	XSetWindowBorder(dpy, win, scheme[SchemeSel][ColBg].pixel);
 	XSetClassHint(dpy, win, &ch);
 
 
-- 
2.37.2

